webapp/app/forms/flows/eligibility_step_chooser.py

```diff
@@ -97,3 +98,18 @@ def determine_prev_step(self, current_step_name, stored_data):
             if possible_previous_step.is_previous_step(current_step_name, stored_data):
                 return possible_previous_step
         return self.steps[self.step_order[0]]
+
+    def _is_step_name_valid(self, step_name):
+        return super()._is_step_name_valid(step_name) or step_name == "first_input_step"
+
+    def _get_possible_redirect(self, step_name, stored_data):
+        if not self._is_step_name_valid(step_name):
+            abort(404)
+        if step_name == 'first_input_step':
+            dbg = self.default_data()
+            if dbg:
+                return dbg[0].name
+            else:
+                return self.determine_next_step(self.first_step.name, stored_data).name
+        return super()._get_possible_redirect(step_name, stored_data)
```

Comment:
I think that shows that the method is not written very nicely, because we can't just use the super method. 

I guess that it would be better if this method would only be an integrator (calling other functions) and all the functionality would be moved into their own operator methods (validity check, find constant step by name, get_redirection_steps).

But this is just fyi &  food for thought 🤔